kong/plugins/prometheus/exporter.lua

```diff
@@ -312,6 +319,11 @@ local should_export_upstream_health_metrics = false
 
 
 local function metric_data(write_fn)
+  if not enabled then
+    kong.log.info("prometheus: no plugin instance enabled, so export endpoint is disabled.")
+    return kong.response.exit(404, { message = "No instance" })
```

Comment:
Some customers use the Prometheus plugin to monitor non-proxy path metrics that this plugin exposes and this change will break those customers.
I strongly advise not proceeding with this break change.